### PR TITLE
Support of Winsum ZH03B sensor Dust particle sensor in xsns_18_pms5003.ino

### PR DESCRIPTION
**Pull request by FransOv**
Support of Winsen ZH03B Dust particle sensor in xsns_18_pms5003.ino

## Description:
The Winsen ZH03B sensor is quite similar to the Plantower PMS3003 sensor. The main difference is the response in passive mode. With these changes xsns_18_pms5003.ino can support the ZH03B sensor when PMS_MODEL_ZH03X is defined in the configuration file.

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [x ] The pull request is done against the latest development branch
  - [ x] Only relevant files were touched
  - [ x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [ x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.14
  - [ x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino [hunk 1]
```diff
@@ -217,20 +283,34 @@ void PmsSecond(void)                 // Every second
...
+        validread = PmsReadData();  // In active mode the rsponse is identical to the PMS sensors
+      }
+    if (validread) {
+#else
     if (PmsReadData()) {
+#endif  // PMS_MODEL_ZH03X
       Pms.valid = 10;
       if (Settings->pms_wake_interval >= MIN_INTERVAL_PERIOD) {
         PmsSendCmd(CMD_SLEEP);
+        AddLog(LOG_LEVEL_DEBUG_MORE, "ZH03x: Sleep command issued");
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino:313]: Pls remove these log messages as they are generic for all models.

File: tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino [hunk 2]
```diff
@@ -239,6 +319,7 @@ void PmsSecond(void)                 // Every second
         Pms.valid--;
         if (Settings->pms_wake_interval >= MIN_INTERVAL_PERIOD) {
           PmsSendCmd(CMD_READ_DATA);
+          AddLog(LOG_LEVEL_DEBUG_MORE, "ZH03x: Read command issued (2)");
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino:322]: Ditto

File: tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino [hunk 3]
```diff
@@ -217,20 +283,34 @@ void PmsSecond(void)                 // Every second
       // wakeup sensor WARMUP_PERIOD before read interval
       Pms.wake_mode = 1;
       PmsSendCmd(CMD_WAKEUP);
+      AddLog(LOG_LEVEL_DEBUG_MORE, "ZH03x: Wake up command issued");
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino:286]: Ditto

File: tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino [hunk 4]
```diff
@@ -171,6 +184,59 @@ bool PmsReadData(void)
   return true;
 }
 
+bool ZH03ReadDataPassive() // process the passive mode response of the ZH03x sensor
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino:187]: Place this function within #ifdef PMS_MODEL_ZH03X

File: tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino [hunk 5]
```diff
@@ -27,6 +27,8 @@
  * You can either support PMS3003 or PMS5003-7003 at one time. To enable the PMS3003 support
  * you must enable the define PMS_MODEL_PMS3003 on your configuration file.
  * For PMSx003T models that report temperature and humidity define PMS_MODEL_PMS5003T
+ * This module can also support de Winsum ZH03x series of dust particle sensors,
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino:30]: It's ``Winsen``. Not Winsum, places in the Netherlands (yes there are more than one)

File: tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino (hunk 2 above)
Comment (FransOv) [reply tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino]: Done

File: tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino (hunk 3 above)
Comment (FransOv) [reply tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino]: Done

File: tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino [hunk 6]
```diff
@@ -217,20 +283,34 @@ void PmsSecond(void)                 // Every second
       // wakeup sensor WARMUP_PERIOD before read interval
       Pms.wake_mode = 1;
       PmsSendCmd(CMD_WAKEUP);
+      AddLog(LOG_LEVEL_DEBUG_MORE, "ZH03x: Wake up command issued");
     }
     if (Pms.time >= Settings->pms_wake_interval) {
       // sensor is awake and warmed up, set up for reading
       PmsSendCmd(CMD_READ_DATA);
+      AddLog(LOG_LEVEL_DEBUG_MORE, "ZH03x: Read command issued (1)");
```
[the remark this answers is not in the document]
Comment (FransOv) [reply tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino]: Done

File: tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino (hunk 4 above)
Comment (FransOv) [reply tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino:187]: Done

File: tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino (hunk 5 above)
Comment (FransOv) [reply tasmota/tasmota_xsns_sensor/xsns_18_pms5003.ino]: Done, Thank you for the comments.

**Comment by arendst** [comment]
Thx.

**Pull request merged**